### ISSUE TITLE
pk sync: from a non-blobpacked src to a blobpacked dest

### ISSUE THREAD

**Issue by mpl** [perkeep/perkeep#1073]
At the moment, I don't think we have any way to migrate data that is stored in a non non-packed way (localdisk blobserver), to a packed way (blobpacked server).
I think adding an option that could do that, to "pk sync", would be valuable.

P.S: my largest datastore is still non blobpacked, and if I could migrate it to blobpacked, maybe that would help me reproduce #1068

**Comment by bradfitz** [comment]
Well, you can already modify your server config to add a blobpacked storage target at some alternate path (/bs-packed/) and then use pkg sync from the old path to new path. I believe that's how I converted.

But yes, this could be easier and documented.

**Comment by mpl** [comment]
> Well, you can already modify your server config to add a blobpacked storage target at some alternate path (/bs-packed/) and then use pkg sync from the old path to new path. I believe that's how I converted.

No, I think that won't work properly, for the same reason we have #681 open: blobs are enumerated in blob order, so when the dest receives a file schema, it's very likely that it does not have the data blobs for that file yet, therefore it will fail packing it. And you end up with most of your blobs in the loose part of the blobpacked storage.

We need to have a smarter enumeration for that use case, which streams files from the source, instead of enumerating by lexicographic order.

**Comment by mpl** [comment]
Moving to 1.0 because I can't migrate to blobpacked, which is pretty bad imho.

**Comment by mpl** [comment]
https://camlistore-review.googlesource.com/#/c/camlistore/+/16866

**Comment by mpl** [comment]
FWIW, I've just tried running this tool on my data set 
(2018/05/23 02:04:27 index/corpus: stats: 64.995 MiB mem: 249651 blobs (11.800 GiB) (87786 schema (8049 permanode, 5439 file (5324 image), ...)), followed by a 'pk sync', and finally a reindex, and the final result seems to be what's expected, as far as the index/corpus is concerned:
2018/05/23 03:01:57 index/corpus: stats: 65.278 MiB mem: 249652 blobs (11.800 GiB) (87786 schema (8049 permanode, 5439 file (5324 image), ...)
